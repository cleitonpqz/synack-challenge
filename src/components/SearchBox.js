import React, { useState } from 'react';

const SearchBox = () => {
  const [searchText, setSearchText] = useState('');
  const onTextChange = (event) => {
    console.log(event.target.value);
    setSearchText(event.target.value);
  };
  return (
    <div>
      <input type="text" onChange={onTextChange} value={searchText}></input>
    </div>
  );
};

export default SearchBox;
