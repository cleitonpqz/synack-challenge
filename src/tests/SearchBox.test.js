import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchBox from '../components/SearchBox';

it('should render SearchBox correctly', () => {
  const component = renderer.create(<SearchBox />).toJSON();
  expect(component).toMatchSnapshot();
});

it('should changes the text captured on typing', () => {
  const value = 'foo';

  //defining a mock function that should adjust the state of search text value
  const setSearchText = jest.fn();

  const wrapper = shallow(<SearchBox onChange={setSearchText} />);

  /**
   * This function creates a mock function similar to jest.fn while tracking the calls
   * to the object’s method (methodName). So we’re testing to validate whether calling this
   * function actually calls the useState hook (function).
   * */
  const handleChange = jest.spyOn(React, 'useState');
  handleChange.mockImplementation((searchText) => [searchText, setSearchText]);
  wrapper.find('input').simulate('change', {
    target: { value },
  });

  /**
   * If useState is called, when we simulate the change on the input element,
   * we should get a truthy value for the setSearchText function.
   * */
  expect(setSearchText).toBeTruthy();
});
