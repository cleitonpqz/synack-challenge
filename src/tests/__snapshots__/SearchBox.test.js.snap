// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBox correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
